Payjoin-spec.md

```diff
@@ -0,0 +1,182 @@
+```diff
+- Step 1. Sender creates a bitcoin transaction paying the receiver
++ Step 1. Sender creates a PSBT paying the receiver
+
+- This transaction must use segwit for all inputs, and be fully valid and signed. The transaction must be eligible for propagation on the network (but not done so at this stage) 
++ The PSBT must be finalized with witness UTXO information.
+
+- Step 2. Sender gives the "template transaction" to the receiver
++ Step 2. Sender gives the "original PSBT" to the receiver
+
+- This is done via an HTTP POST request, sent to a "[bustapay] url" 
++ This is done via an HTTP POST request, sent to a "[payjoin] url" 
+
+Step 3. Receiver processes the transaction and returns a partially signed coinjoin
+
+- The receiver validates the transaction, and pays himself. The receiver then adds one or more of his own inputs (known as the contributed inputs) and (optionally) increases the output that pays himself (generally by the sum of the contributed inputs). Doing so creates a partial transaction, which the receiver returns to the sender. It is called such as it requires the sender to re-sign his own inputs. 
++ The receiver extracts the transaction, optionally verifying it is valid. (by calling testmempoolaccept on the extracted transaction)
++ The receiver extracts the feerate, add one of his input and decrease the change output of the payer by the fee his input added such that the feerate is the same as the "orginal PSBT".
+```
+
+The rest of the process is similar to bustapay.
+We also support the use of raw transaction instead of PSBT, but strongly advise against it.
+
+We will refer to the sender's PSBT by the name of `original PSBT` and the receiver's payjoin proposal as `payjoin PSBT`.
+
+## Rational
+
+Here are the rational for using PSBT instead of raw transactions, and why the receiver should be responsible to bump the fee of the payjoin transaction.
+
+### Respecting the minimum relay fee policy
+To be properly relayed, a Bitcoin transaction need to pay at least 1 satoshi per virtual byte.
+When fee are low, the original transaction is already 1 satoshi per virtual byte, so if the merchant add his own input, he need to make sure the fee are increased such that the rate does not drop below 1 satoshi per virtual byte.
+
+### Preventing mempool replacement
+A safe way to implement payjoin, is for both the sender and receiver to try broadcasting the original transaction at some fixed interval period regardless of the state of the payjoin.
+
+If the receiver was not properly adding fees to the payjoin transaction, the original transaction would end up replacing the payjoin transaction in the mempool.
+
+### Receiver does not need to be a full node
+Because the receiver need to bump the keep the same fee rate than the original PSBT, it needs the input's UTXO information to know what is the original fee rate. Without PSBT, light wallets like Wasabi Wallet would not be able to receive a payjoin transaction.
```

Comment:
`Because the receiver needs to bump the fee to keep the same fee rate as the original PSBT, it needs the input's UTXO information to know what is the original fee rate. Without PSBT, light wallets like Wasabi Wallet would not be able to receive a payjoin transaction.`